packages/libs/deer/json/src/lib.rs

```diff
@@ -445,43 +468,110 @@ impl<'a, 'de> deer::ArrayAccess<'de> for ArrayAccess<'a> {
 
 #[must_use]
 struct ObjectAccess<'a> {
+    dirty: bool,
+    length: usize,
+
     inner: Map<String, Value>,
     context: &'a Context,
+
+    remaining: Option<usize>,
 }
 
 impl<'a> ObjectAccess<'a> {
-    const fn new(map: Map<String, Value>, context: &'a Context) -> Self {
+    fn new(map: Map<String, Value>, context: &'a Context) -> Self {
```

Comment:
Instead of a `set_bounded` and having two errors, which basically ensures, that the parameter is passed first, why don't we just pass it here?